manifests/plugin.pp

```diff
@@ -79,13 +79,13 @@
       unless      => $unless_install,
     }
   } elsif $ensure == 'absent' {
-    $docker_plugin_remove_flags = docker_plugin_remove_flags( {
+    $docker_plugin_remove_flags = docker_plugin_remove_flags({
         plugin_name => $plugin_name,
         force_remove => $force_remove,
       }
     )
 
-    $exec_rm   = "${docker_command} rm ${docker_plugin_remove_flags}"
+    $exec_rm   = [$docker_command, 'rm', $docker_plugin_remove_flags]
     $onlyif_rm = "${docker_command} ls --format='{{.PluginReference}}' | grep -w ${plugin_name}"
```

Comment:
This is an interesting one. `$onlyif_rm` should be broken out because both `$docker_command` and `$plugin_name` come from parameters.

With exec, both `onlyif` and `unless` accept a `String` or an `Array[Array[String]]`.

For the latter it would look like this

```
onlyif => [ ['my', 'command', 'and', 'args'] ]

```

When it is evaluated exec looks to see if all commands are truthy (e.g you may have multiple inner arrays for different commands).

The tricky part here is that when using an array of args, exec will disregard any metacharacters (right word??). In this case it would ignore the pipe and everything to the right of it.

There may be an example of how to handle this in apt or mysql. Alternatively we could just clean the strings with shell_escape or something similar.

Reply:
Talked with @david22swan about this cases. I believe we just agreed to not touch anything with pipes because of the reason mentioned. I did try some variations arraying and segmenting like:
`$onlyif_rm = ["$docker_command, 'ls', "--format='{{.PluginReference}}' | grep -w ${plugin_name}"]`

Just for the sake of testing whether the problem was separating the pipe from the rest of the arguments but it did not work. Currently any command with a grep is untouched in this and every other PR afaik.

Reply:
It should be fine so long as the command is sanitized in some other way.
In this case the only variable part, `docker_command` should be given a shell escape as while it has a default value of `docker` it is able to be manually set.
It's a difficult one as it is explicitly meant to be a variable for overriding what the command does....

Reply:
..... Could add a regex check to it to make sure that the command always includes docker at the end with no white spaces in front of it, i.e. that it's always a path to the docker command